packages/zendesk-support-adapter/src/client/connection.ts

```diff
@@ -41,20 +41,39 @@ export const validateCredentials = async ({ credentials, connection }: {
   return credentials.subdomain
 }
 
+const usernamePasswordAuthParamsFunc = (
+  { username, password }: UsernamePasswordCredentials
+): AuthParams => ({
+  auth: {
+    username,
+    password,
+  },
+  headers: {
+    'X-Zendesk-Marketplace-Name': MARKETPLACE_NAME,
+    'X-Zendesk-Marketplace-Organization-Id': MARKETPLACE_ORG_ID,
+    'X-Zendesk-Marketplace-App-Id': MARKETPLACE_APP_ID,
+  },
+})
+
+const accessTokenAuthParamsFunc = (
+  { accessToken }: OauthAccessTokenCredentials
+): AuthParams => ({
+  headers: {
+    Authorization: `Bearer ${accessToken}`,
+    'X-Zendesk-Marketplace-Name': MARKETPLACE_NAME,
+    'X-Zendesk-Marketplace-Organization-Id': MARKETPLACE_ORG_ID,
+    'X-Zendesk-Marketplace-App-Id': MARKETPLACE_APP_ID,
+  },
```

Comment:
nit: since the last 3 headers are all constants, may be worth just having an object with these 3 as a constant somewhere to avoid duplicating the header names. then this can become
```suggestion
  headers: {
    Authorization: `Bearer ${accessToken}`,
    ...APP_MARKETPLACE_HEADERS,
  },
```
(or whatever name you choose for that const)